src/test/java/com/google/api/generator/engine/writer/JavaWriterVisitorTest.java

```diff
@@ -205,6 +209,111 @@ public void writeVariableExpr_scopedStaticFinalDecl() {
     assertEquals(writerVisitor.write(), "public static final boolean x");
   }
 
+  @Test
+  public void writeNormalBlockComment() {
+    String content = "this is a test comment";
+    BlockComment blockComment = BlockComment.builder().setComment(content).build();
+    String expected = "/** this is a test comment */\n";
+    blockComment.accept(writerVisitor);
+    assertEquals(writerVisitor.write(), expected);
+  }
+
+  @Test
+  public void writeBadBlockComment() {
+    String content =
+        "A super long long long long long long long */ long long long long long long long comment";
+    assertThrows(
+        RuntimeException.class,
```

Comment:
On reading this again, I think we should have a nested dedicated exception class in the formatter singleton. Could we please add this?

Reply:
sure, added a public `FormatException` nested class in JavaFormatter and updated the test.
It is set to be public so that we can directly call it (assert.throws) from `BlockCommentTest` bad test case.

Reply:
If the exception does not need to be accessed by package-external classes, could we please we make it protected and add a `@VisibleForTesting` annotation?

Reply:
Yes, it is in the same package, changed it to protected and added the annotation. Thanks for the advice